### PR TITLE
Bug Issues mentee validation and email validation

### PR DESCRIPTION
**Pull request by sumanvpacewisdom**

### CODE REVIEW

File: src/database/queries/mentorExtension.js [hunk 1]
```diff
@@ -310,4 +310,28 @@ module.exports = class MentorExtensionQueries {
...
+			const validMentors = []
+			const invalidMentors = []
+			userIds.forEach((userId) => {
+				const mentor = mentorMap.get(userId)
+				if (mentor) {
+					validMentors.push(mentor)
+				} else {
+					invalidMentors.push(userId)
+				}
+			})
```
Comment (nevil-mathew) [inline src/database/queries/mentorExtension.js:331]: We cannot have these here. Functions inside queriers should return DB response as it is.

File: src/database/queries/userExtension.js [hunk 2]
```diff
@@ -298,4 +298,28 @@ module.exports = class MenteeExtensionQueries {
...
+			const validMentees = []
+			const invalidMentees = []
+			userIds.forEach((userId) => {
+				const mentee = menteeMap.get(userId)
+				if (mentee) {
+					validMentees.push(mentee)
+				} else {
+					invalidMentees.push(userId)
+				}
+			})
```
Comment (nevil-mathew) [inline src/database/queries/userExtension.js:319]: We cannot have these here. Functions inside queriers should return DB response as it is.

File: src/constants/common.js [hunk 3]
```diff
@@ -152,6 +152,8 @@ module.exports = {
 	},
 	PLATFORMS_REGEX: /https:\/\/(?:meet|call|us\d{2}web)\.(\w+)\.com/,
 	ZOOM_REGEX: /https:\/\/(?:meet|call|us\d{2}web|zoom)\.(\w+)\.us\/j\/(\d+)\?/,
+	EMAIL_REGEX:
+		/^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/,
```
Comment (nevil-mathew) [inline src/constants/common.js:156]: Can we use the isValidEmail from utils?

File: src/services/sessions.js [hunk 4]
```diff
@@ -85,6 +85,14 @@ module.exports = class SessionsHelper {
 			const mentorIdToCheck = bodyData.mentor_id || loggedInUserId
 			const isSessionCreatedByManager = !!bodyData.mentor_id
 
+			if (bodyData.type == common.SESSION_TYPE.PRIVATE && menteeIdsToEnroll.length === 0) {
+				return responses.failureResponse({
+					message: 'MENTEES_NOT_FOUND',
```
Comment (nevil-mathew) [inline src/services/sessions.js:90]: Is the message added in en.json?

File: src/services/sessionInvite.js [hunk 5]
```diff
@@ -441,22 +442,28 @@ module.exports = class UserInviteHelper {
 					session.status = 'Invalid'
 					session.statusMessage = this.appendWithComma(session.statusMessage, ' Only 6 mentees are allowed')
```
Comment (nevil-mathew) [inline src/services/sessionInvite.js:443]: Why is this message hardcoded.
Isn't mentee count configurable ?

File: src/services/mentees.js [hunk 6]
```diff
@@ -1339,6 +1339,80 @@ module.exports = class MenteesHelper {
...
+
+	/**
+	 * @description 							- check if mentee is accessible based on user's saas policy.
+	 * @method
+	 * @name checkIfMenteeIsAccessible
+	 * @param {Number} userId 					- User id.
+	 * @param {Array}							- Session data
+	 * @param {Boolean} isAMentor 				- user mentor or not.
+	 * @returns {JSON} 							- List of filtered sessions
+	 */
```
Comment (nevil-mathew) [inline src/services/mentees.js:1351]: Fix the JS doc

File: src/services/sessions.js [hunk 7]
```diff
@@ -109,6 +117,38 @@ module.exports = class SessionsHelper {
...
+				const menteesDetailsInMentor = await mentorExtensionQueries.getMentorsExtensions(menteeIdsToEnroll)
+				const invalidMentorId =
+					menteesDetailsInMentor.invalidMentors.length === 0 ? [] : menteesDetailsInMentor.invalidMentors
+				const menteesDetailsInMentee = await menteeExtensionQueries.getMenteesExtensions(invalidMentorId)
+				if (
+					(menteesDetailsInMentor.validMentors.length === 0) &
+					(menteesDetailsInMentee.validMentees.length === 0)
+				) {
+					return responses.failureResponse({
+						message: 'INVALID_PERMISSION',
```
Comment (nevil-mathew) [inline src/services/sessions.js:131]: Is this message correct?

**Review by nevil-mathew — CHANGES_REQUESTED** [review]

File: src/database/queries/mentorExtension.js (hunk 1 above)
Comment (sumanvpacewisdom) [reply src/database/queries/mentorExtension.js:331]: ok can i use it in create session as a function ? @nevil-mathew

File: src/database/queries/userExtension.js (hunk 2 above)
Comment (sumanvpacewisdom) [reply src/database/queries/userExtension.js:319]: ok can i use it in create session as a function ? @nevil-mathew

File: src/services/sessions.js (hunk 4 above)
Comment (sumanvpacewisdom) [reply src/services/sessions.js:90]: Yes , it is added , i am using the existing messages only .

File: src/services/sessionInvite.js (hunk 5 above)
Comment (sumanvpacewisdom) [reply src/services/sessionInvite.js:443]: Do i have to make it configurable @nevil-mathew ?

File: src/services/sessions.js (hunk 7 above)
Comment (sumanvpacewisdom) [reply src/services/sessions.js:131]: @nevil-mathew Yes .

File: src/constants/common.js (hunk 3 above)
Comment (sumanvpacewisdom) [reply src/constants/common.js:156]: We can not add in utils @nevil-mathew

File: src/services/sessionInvite.js (hunk 5 above)
Comment (nevil-mathew) [reply src/services/sessionInvite.js:443]: Yes

File: src/database/queries/mentorExtension.js (hunk 1 above)
Comment (nevil-mathew) [reply src/database/queries/mentorExtension.js:331]: You can move this to the appropriate function, but this has to be moved from queries.

File: src/database/queries/userExtension.js (hunk 2 above)
Comment (nevil-mathew) [reply src/database/queries/userExtension.js:319]: You can move this to the appropriate place, but this has to be moved from queries.

File: src/services/sessions.js [hunk 8]
```diff
@@ -2459,6 +2555,8 @@ module.exports = class SessionsHelper {
 				})
 			}
 
+			console.log('csvData', csvData)
```
Comment (nevil-mathew) [inline src/services/sessions.js:2558]: remove this console.log

File: src/services/sessions.js [hunk 9]
```diff
@@ -4,7 +4,8 @@ const moment = require('moment-timezone')
 const httpStatusCode = require('@generics/http-status')
 const apiEndpoints = require('@constants/endpoints')
 const common = require('@constants/common')
-
+const MentorExtension = require('@database/models/index').MentorExtension
+const MenteeExtension = require('@database/models/index').UserExtension
```
Comment (nevil-mathew) [inline src/services/sessions.js:8]: Remove this. We should only import queries to service files.

File: src/services/sessions.js (hunk 7 above)
Comment (nevil-mathew) [reply src/services/sessions.js:133]: Change this message. It's not invalid permissions if the mentees added were not found.
Other places, if used, we can change later.

File: src/services/sessions.js [hunk 10]
```diff
@@ -2578,4 +2672,46 @@ module.exports = class SessionsHelper {
...
+					invalidMentors.push(userId)
+				}
+			})
+			return { validMentors, invalidMentors }
+		} catch (error) {
+			throw error
+		}
+	}
+
+	static async getMenteesExtensions(userIds) {
```
Comment (nevil-mathew) [inline src/services/sessions.js:2697]: Can we rename it to 'validateMenteeExtensions'

File: src/services/sessions.js [hunk 11]
```diff
@@ -2578,4 +2672,46 @@ module.exports = class SessionsHelper {
 			}
 		}
 	}
+
+	static async getMentorsExtensions(userIds) {
```
Comment (nevil-mathew) [inline src/services/sessions.js:2676]: Can we rename it to 'validateMentorExtensions'

**Review by nevil-mathew — CHANGES_REQUESTED** [review]

**Review by nevil-mathew — APPROVED** [review]

**Pull request merged**